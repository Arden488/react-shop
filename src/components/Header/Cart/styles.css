.header-cart {
  text-align: right;
}

.header-cart a {
  color: #fff;
}

.header-cart span {
  display: inline-block;
  vertical-align: middle;
  margin-left: 10px;
}

.header-cart svg {
  display: inline-block;
  vertical-align: middle;
}