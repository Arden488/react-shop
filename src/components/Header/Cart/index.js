import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import feather from 'feather-icons';

import './styles.css';

class Cart extends Component {
  componentDidMount() {
    feather.replace();
  }

  render() {
    return (
      <div className="header-cart">
        <Link to="/cart">
          <i data-feather="shopping-cart" />
          <span>0 руб.</span>
        </Link>
      </div>
    );
  }
}

export default Cart;
