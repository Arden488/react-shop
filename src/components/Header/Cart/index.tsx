import * as feather from 'feather-icons';
import * as React from 'react';
import { Link } from 'react-router-dom';

import Dropdown from '../../Dropdown';
import DropdownContent from '../../Dropdown/DropdownContent';

import './styles.css';

class Cart extends React.Component {
  public componentDidMount() {
    feather.replace();
  }

  public render() {
    return (
      <div className="header-cart">
        <Dropdown>
          <button>
            <i data-feather="shopping-cart" />
            <span>0 руб.</span>
          </button>
          <DropdownContent>
            <p>...</p>
            <Link to="/cart">Proceed to checkout</Link>
          </DropdownContent>
        </Dropdown>
      </div>
    );
  }
}

export default Cart;
