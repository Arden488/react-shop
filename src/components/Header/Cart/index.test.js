/* global it, expect, beforeEach */

import React from 'react';
import { shallow } from 'enzyme';

import Cart from './index';

let component = null;

beforeEach(() => {
  component = shallow(<Cart />);
});

it('renders without crashing', () => {
  expect(component.exists()).toEqual(true);
});
