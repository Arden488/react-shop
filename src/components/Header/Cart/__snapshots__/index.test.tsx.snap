// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="header-cart"
>
  <Dropdown>
    <button>
      <i
        data-feather="shopping-cart"
      />
      <span>
        0 руб.
      </span>
    </button>
    <DropdownContent>
      <p>
        ...
      </p>
      <Link
        replace={false}
        to="/cart"
      >
        Proceed to checkout
      </Link>
    </DropdownContent>
  </Dropdown>
</div>
`;
